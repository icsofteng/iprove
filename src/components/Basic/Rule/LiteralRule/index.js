import React, { Component } from 'react'
import styles from '../styles.scss'

class LiteralRule extends Component {
  componentDidMount() {
    if (this.ref) {
      this.ref.focus()
    }
  }
  
  render() {
    return (
      <input
        type="text"
        value={this.props.value}
        onChange={(event) => {
          const upper = event.target.value.toUpperCase()
          this.props.addConstants(upper)
          this.props.updateValue([...this.props.path, "value"], upper)
        }}
        className={styles.ruleInput}
        ref={(ref) => this.ref = ref}
      />
    )
  }
}
  
export default LiteralRule
