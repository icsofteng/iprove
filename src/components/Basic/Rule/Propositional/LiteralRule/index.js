import React, { Component } from 'react'
import styles from '../../styles.scss'

class LiteralRule extends Component {
  componentDidMount() {
    if (this.ref) {
      this.ref.focus()
    }
  }

  render() {
    return (
      <input
        type="text"
        value={this.props.value}
        onChange={(event) => {
          const upper = event.target.value.toUpperCase()
          this.props.addIdentifiers({ value: upper, varType: "Bool" })
          this.props.updateValue([...this.props.path, "value"], upper)
          this.props.updateValue([...this.props.path, "varType"], "Bool")
        }}
        className={styles.ruleInput}
        ref={(ref) => this.ref = ref}
      />
    )
  }
}

export default LiteralRule
