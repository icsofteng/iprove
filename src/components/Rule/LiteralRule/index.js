import React from 'react'

import styles from '../styles.scss'

const LiteralRule = (props) =>
  <input
    type="text"
    value={props.value}
    onChange={(event) => {
      props.addConstant(event.target.value)
      props.updateValue([...props.path, "value"], event.target.value)
    }}
    className={styles.ruleInput}
  />

export default LiteralRule
