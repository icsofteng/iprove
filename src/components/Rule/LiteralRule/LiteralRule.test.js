import React from 'react'
import LiteralRule from '.'


test('Literal undefined test', () => {
  const wrapper = shallow(<LiteralRule type={'literal'} path={[]}/>)
  expect(wrapper).toMatchSnapshot()
})

test('Literal p test', () => {
  const wrapper = shallow(<LiteralRule type={'literal'} value={'p'} path={[]}/>)
  expect(wrapper).toMatchSnapshot()
})