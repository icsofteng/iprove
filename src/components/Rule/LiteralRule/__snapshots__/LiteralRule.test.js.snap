// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Literal p test 1`] = `
<input
  className="ruleInput"
  onChange={[Function]}
  type="text"
  value="p"
/>
`;

exports[`Literal undefined test 1`] = `
<input
  className="ruleInput"
  onChange={[Function]}
  type="text"
/>
`;
